src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                }
+
+                // OrderBy is not supported by foreach.
+                return false;
+            }
+
+            private bool TryProcessSelectClause(
+                SelectClauseSyntax selectClause,
+                QueryContinuationSyntax queryContinuation,
+                ProcessingState processingState)
+            {
+                // TODO semantic check for selectClause.Expression
+                // End of the queue
+                if (queryContinuation == null)
+                {
+                    processingState.AddStatementProcessingMethod(
+                        s => WrapWithBlock(processingState.LeafExpressionCreationMethod(selectClause.Expression)));
+                    return true;
+                }
+                else
+                {
+                    SymbolInfo symbolInfo = _semanticModel.GetSymbolInfo(queryContinuation, _cancellationToken);
+                    if (symbolInfo.Symbol == null ||
+                        symbolInfo.Symbol is IMethodSymbol methodSymbol &&
+                        methodSymbol.Parameters.Length == 1)
+                    {
+                        processingState.AddVariableName(queryContinuation.Identifier.ValueText);
+                        processingState.AddStatementProcessingMethod(
+                            s => AddToBlockTop(
+                                 SyntaxFactory.LocalDeclarationStatement(
+                                    SyntaxFactory.VariableDeclaration(
+                                        VarNameIdentifier,
+                                        SyntaxFactory.SingletonSeparatedList(
+                                            SyntaxFactory.VariableDeclarator(
+                                                queryContinuation.Identifier,
+                                                argumentList: null,
+                                                SyntaxFactory.EqualsValueClause(selectClause.Expression))))), s));
+
+                        return TryProcessQueryBody(queryContinuation.Body, processingState);
+                    }
+                }
+
+                return false;
+            }
+
+            private bool TryProcessWhereClause(WhereClauseSyntax whereClause, ProcessingState processingState)
+            {
+                if (IsValidWhereClauseSymbol(whereClause))
+                {
+                    processingState.AddStatementProcessingMethod(
+                         s => SyntaxFactory.Block(
+                             SyntaxFactory.IfStatement(
+                                 ProcessCondition((whereClause).Condition),
+                                 s)));
+
+                    return true;
+                }
+
+                return false;
+            }
+
+            private bool IsValidWhereClauseSymbol(WhereClauseSyntax whereClause)
+            {
+                if (!TryGetMethodSymbol(whereClause, out var methodSymbol))
```

Comment:
>if (!TryGetMethodSymbol(whereClause, out var methodSymbol)) [](start = 16, length = 59)

Why are we not checking method's name?

Reply:
agreed to skip verification of query clauses for now

---
In reply to: [177199664](https://github.com/dotnet/roslyn/pull/25362#discussion_r177199664) [](ancestors = 177199664)